openmdao/drivers/genetic_algorithm_driver.py

```diff
@@ -281,6 +321,34 @@ def objective_callback(self, x, icase):
                 obj = val
                 break
 
+            # Parameters of the penalty method
+            penalty = self.options['penalty_parameter']
+            exponent = self.options['penalty_exponent']
+
+            if penalty == 0:
+                fun = obj
+            else:
+                constraint_vals = list()
+                constraint_violations = np.array([])
+                for name, val in iteritems(self.get_constraint_values()):
+                    constraint_vals.append(val)
+                    con = self._cons[name]
+                    # The not used fields will either None or a very large number
+                    if (con['lower'] is not None) and np.isfinite(con['lower']):
+                        diff = val - con['lower']
+                        violation = np.array([0. if d >= 0 else abs(d) for d in diff])
+                    elif (con['upper'] is not None) and np.isfinite(con['upper']):
+                        diff = val - con['upper']
+                        violation = np.array([0. if d <= 0 else abs(d) for d in diff])
+                    elif (con['equals'] is not None) and np.isfinite(con['equals']):
+                        diff = val - con['equals']
+                        violation = np.absolute(diff)
+                    else:
```

Comment:
Don't think you need the else branch, since the constraint metadata should be checked during add_constraint (and if we don't check for cons without lower|upper|equals there, then we should.)